apps/web/src/actions/experiments/create.ts

```diff
@@ -56,23 +56,32 @@ export const createExperimentAction = withDocument
       )
     }
 
-    const experiment = await createExperiment({
-      name,
-      document: ctx.document,
-      commit: ctx.commit,
-      evaluations,
-      dataset,
-      parametersMap,
-      datasetLabels,
-      fromRow,
-      toRow,
-      workspace: ctx.workspace,
-    }).then((r) => r.unwrap())
+    const experiments = await Promise.all(
+      variants.map((variant) =>
+        createExperiment({
+          name: variant.name,
+          customPrompt: variant.prompt,
+          document: ctx.document,
+          commit: ctx.commit,
+          evaluations,
+          dataset,
+          parametersMap,
+          datasetLabels,
+          fromRow,
+          toRow,
+          workspace: ctx.workspace,
+        }).then((r) => r.unwrap()),
+      ),
+    )
 
-    await startExperiment({
-      workspace: ctx.workspace,
-      experimentUuid: experiment.uuid,
-    }).then((r) => r.unwrap())
+    await Promise.all(
+      experiments.map((experiment) =>
+        startExperiment({
+          workspace: ctx.workspace,
+          experimentUuid: experiment.uuid,
+        }).then((r) => r.unwrap()),
+      ),
+    )
 
-    return experiment
+    return experiments
```

Comment:
missing tests for this action, also move the whole logic to a service and make it atomic: open a transaction that creates all experiments and starts them in a single operation.